hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1431,6 +1431,8 @@ private void startServiceThreads() throws IOException{
       conf.getInt("hbase.master.executor.meta.serverops.threads", 5));
    this.executorService.startExecutorService(ExecutorType.M_LOG_REPLAY_OPS,
       conf.getInt("hbase.master.executor.logreplayops.threads", 10));
+   this.executorService.startExecutorService(ExecutorType.MASTER_SNAPSHOT_OPERATIONS,
```

Comment:
IIRC we only allow one snapshot at the same time? No?

Reply:
Different table can snapshot at the same time , for the given table, only one snapshot can be taking (Have a tableLock to ensure that).